feat: migrate the rest of public site to new backend

ludtkemorgan opened the pull request:
# Pull Request Template

## Issue Overview

This PR addresses #3614 and #3616 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR has the rest of the changes required on the public site to get it connected to the new Prisma based backend. It includes all remaining flows so all pages on the Public site should work.
This includes but is not limited to:
- Application submission
- User creation
- Login
- application dashboard

This PR also has some changes to the backend code to cover bugs found when integrating with the frontend

Note: this has a lot of file changes in the partner site. This is due to the intermingling of code between the two front-end sites. Most of the changes on partner site are needed for the long term connection for the Partner site, however, there are some temporary changes that can be deleted when the partner site is connected. Specifically all of the `X as unknown as X`. These are just in place while both old and new types are in the code base.

## How Can This Be Tested/Reviewed?

Do a full e2e test of the public front-end

The cypress tests work locally but failing on the CI job. Trying to dig through that issue

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

ColinBuyck commented:
Still reviewing but found a small bug in populating the state field 

https://github.com/bloom-housing/bloom/assets/53269332/f2a066d8-19fd-4338-a5b2-d1169a2d3d62

ColinBuyck commented:
It seems that this PR doesn't handle the application link saving that we've discussed before. In other words, users jumping to the middle of a new application via url and then leading them back with the message below: 
 <img width="534" alt="Screenshot 2023-11-14 at 11 00 00 AM" src="https://github.com/bloom-housing/bloom/assets/53269332/6a1484b4-0cdb-446c-b219-a040a6c033a8">

But it doesn't seem covered here since it immediately errors out:
https://github.com/bloom-housing/bloom/assets/53269332/c4f53ca8-9e34-456e-830b-309a1125d142

ColinBuyck commented:
The "See Listing" button on the My Applications page is not functioning (takes me to http://localhost:3000/listing/undefined/undefined)
<img width="608" alt="Screenshot 2023-11-14 at 11 43 06 AM" src="https://github.com/bloom-housing/bloom/assets/53269332/3faa492e-f00e-49fc-9e3c-956689884cec">

ColinBuyck commented:
Any account setting updates seem to be erroring out with the following error:
<img width="784" alt="Screenshot 2023-11-14 at 11 46 28 AM" src="https://github.com/bloom-housing/bloom/assets/53269332/6258f6f2-f98b-4343-9d14-4d98e9f1c035">

YazeedLoonat commented on sites/public/src/pages/api/adapter/[...backendUrl].ts:
```diff
@@ -41,15 +41,18 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
     res.statusMessage = response.statusText
     res.status(response.status).json(response.data)
+
+    console.log("response headers", response.headers)
```
console.log that made it through

YazeedLoonat reviewed (commented):
Hey Morgan looks good, a few small code things and 1 more general question:

It looks like we had to add back in jurisdictionName for a few endpoints, do you think its better to take that approach than to change what the front end is sending to just sending the jurisdiction id?

ludtkemorgan commented on sites/partners/src/components/listings/ListingFormActions.tsx:
```diff
@@ -15,15 +15,14 @@ import {
-  ListingEventType,
-  ListingStatus,
-} from "@bloom-housing/backend-core/types"
+import { ListingEventType, ListingStatus } from "@bloom-housing/backend-core/types"
```
This is just here because it's on the partner site and it hasn't been tackled yet for this PR. I tried to do as little editing of the partner site as possible, but I ended up touching a decent amount because of all of the shared-helpers code

ludtkemorgan commented on backend_new/src/main.ts:
```diff
@@ -9,8 +9,8 @@ async function bootstrap() {
       process.env.NODE_ENV === 'development'
-        ? ['error', 'warn', 'log']
-        : ['error', 'warn', 'log'],
+        ? ['error', 'warn', 'log', 'debug']
```
I removed the debug and log from non-development. I added it for testing purposes and don't need it anymore

YazeedLoonat reviewed (approved):
LGTM

ludtkemorgan commented:
@ColinBuyck I have fixed the "see listing" issue. 

The jumping between pages on the application flow is happening for me on the main branch as well. I think in a deployed environment it just handles the error differently.

I have also fixed the state issue. However every once and a while it doesn't work if you don't select the suggested address. Can you double check to see if it is happening to you?

ColinBuyck reviewed (approved):
LGTM 🧦 Thanks for making those updates!
